extensions/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseProcessor.java

```diff
@@ -0,0 +1,325 @@
+import io.quarkus.liquibase.LiquibaseMongodbFactory;
+import io.quarkus.liquibase.runtime.LiquibaseMongodbBuildTimeConfig;
+import io.quarkus.liquibase.runtime.LiquibaseMongodbConfig;
+import io.quarkus.liquibase.runtime.LiquibaseMongodbRecorder;
+import io.quarkus.mongodb.runtime.MongodbConfig;
+import liquibase.change.Change;
+import liquibase.change.DatabaseChangeProperty;
+import liquibase.change.core.CreateProcedureChange;
+import liquibase.change.core.CreateViewChange;
+import liquibase.change.core.LoadDataChange;
+import liquibase.change.core.SQLFileChange;
+import liquibase.changelog.ChangeLogParameters;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.exception.LiquibaseException;
+import liquibase.parser.ChangeLogParser;
+import liquibase.parser.ChangeLogParserFactory;
+import liquibase.resource.ClassLoaderResourceAccessor;
+
+class LiquibaseProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(LiquibaseProcessor.class);
+
+    private static final String LIQUIBASE_BEAN_NAME_PREFIX = "liquibase_";
+
+    private static final DotName DATABASE_CHANGE_PROPERTY = DotName.createSimple(DatabaseChangeProperty.class.getName());
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(Feature.LIQUIBASE_MONGODB);
+    }
+
+    @BuildStep
+    public SystemPropertyBuildItem disableHub() {
+        // Don't block app startup with prompt:
+        // Do you want to see this operation's report in Liquibase Hub, which improves team collaboration?
+        // If so, enter your email. If not, enter [N] to no longer be prompted, or [S] to skip for now, but ask again next time (default "S"):
+        return new SystemPropertyBuildItem("liquibase.hub.mode", "off");
+    }
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    @Record(STATIC_INIT)
+    void nativeImageConfiguration(
+            LiquibaseMongodbRecorder recorder,
+            LiquibaseMongodbBuildTimeConfig liquibaseBuildConfig,
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<ReflectiveClassBuildItem> reflective,
+            BuildProducer<NativeImageResourceBuildItem> resource,
+            BuildProducer<ServiceProviderBuildItem> services,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized,
+            BuildProducer<NativeImageResourceBundleBuildItem> resourceBundle) {
+
+        runtimeInitialized.produce(new RuntimeInitializedClassBuildItem(liquibase.diff.compare.CompareControl.class.getName()));
+
+        reflective.produce(new ReflectiveClassBuildItem(false, true, false,
+                liquibase.change.AbstractSQLChange.class.getName(),
+                liquibase.database.jvm.JdbcConnection.class.getName()));
+
+        reflective.produce(new ReflectiveClassBuildItem(true, true, true,
+                liquibase.parser.ChangeLogParserCofiguration.class.getName(),
```

Comment:
any reason for the FQCNs here?

Reply:
Same reason as for your other comment.

Reply:
Well, in fact if I use simple name this will ends up with tens of imports. I think that's why it has been done like this in the liquibase extension. Maybe using String would be better as it would avoid loading the class.

Reply:
Howerver, when using String, you can ends up registering for reflection classes that didn't exist anymore as you'll not be notified when they'll deleted.

Reply:
maybe  @geoand have an advice on this ?

Reply:
personally, i'd rather have imports i can collapse away but don't hold up the merge just for this.

Reply:
It should be just fine the way you have it.